service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/status/service/EhrStatusService.java

```diff
@@ -0,0 +1,188 @@
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import lombok.AllArgsConstructor;
+import uk.nhs.adaptors.gp2gp.ehr.EhrExtractStatusRepository;
+import uk.nhs.adaptors.gp2gp.ehr.model.EhrExtractStatus;
+import uk.nhs.adaptors.gp2gp.ehr.status.model.EhrStatus;
+import uk.nhs.adaptors.gp2gp.ehr.status.model.FileStatus;
+import uk.nhs.adaptors.gp2gp.ehr.status.model.MigrationStatus;
+
+
+@Service
+@AllArgsConstructor(onConstructor = @__(@Autowired))
+public class EhrStatusService {
+
+    private EhrExtractStatusRepository ehrExtractStatusRepository;
+
+    public Optional<EhrStatus> getEhrStatus(String conversationId) {
+
+        Optional<EhrExtractStatus> extractStatusOptional = ehrExtractStatusRepository.findByConversationId(conversationId);
+
+        if (extractStatusOptional.isEmpty()) {
+            return Optional.empty();
+        }
+
+        EhrExtractStatus ehrExtractStatus = extractStatusOptional.get();
+        List<EhrExtractStatus.EhrReceivedAcknowledgement> receivedAcknowledgements = getAckModel(ehrExtractStatus);
+        List<EhrStatus.AttachmentStatus> attachmentStatusList = getAttachmentStatusList(ehrExtractStatus, receivedAcknowledgements);
+
+        return Optional.of(
+            EhrStatus.builder()
+                .attachmentStatus(attachmentStatusList)
+                .migrationLog(receivedAcknowledgements)
+                .migrationStatus(evaluateMigrationStatus(ehrExtractStatus, attachmentStatusList))
+                .originalRequestDate(ehrExtractStatus.getCreated())
+                .build());
+    }
+
+    private List<EhrExtractStatus.EhrReceivedAcknowledgement> getAckModel(EhrExtractStatus ehrExtractStatus) {
+
+        Optional<EhrExtractStatus.AckHistory> ackHistoryOptional = Optional.ofNullable(ehrExtractStatus.getAckHistory());
+
+        return ackHistoryOptional.map(EhrExtractStatus.AckHistory::getAcks).orElse(new ArrayList<>());
+    }
+
+    private MigrationStatus evaluateMigrationStatus(EhrExtractStatus record, List<EhrStatus.AttachmentStatus> attachmentStatusList) {
+
+        var ackPendingOptional = Optional.ofNullable(record.getAckPending());
+        var ackToRequestorOptional = Optional.ofNullable(record.getAckToRequester());
+        var errorOptional = Optional.ofNullable(record.getError());
+        var receivedAcknowledgementOptional = Optional.ofNullable(record.getEhrReceivedAcknowledgement());
+
+        if (
+            ackPendingOptional
+                .map(ackPending -> ackPending.getTypeCode().equals("AE"))
+                .orElse(false)
+
+                && ackToRequestorOptional
+                .map(ackToRequester -> ackToRequester.getTypeCode().equals("AE"))
+                .orElse(false)
+
+                && errorOptional.isPresent()) {
+
+            return FAILED_NME;
+        } else if (
+            ackPendingOptional
+                .map(ackPending -> ackPending.getTypeCode().equals("AA"))
```

Comment:
AA and AE should be replaced with a class specific variable. 
It is not clear what they stand for.